Document Service Maturity (#12)
appuio/cloud-kubernetes-docs: docs/modules/ROOT/pages/references/glossary.adoc

= Glossary

////
You can link these terms using xrefs and their IDs, for example:

>   xref:reference/glossary.adoc#service-instance[Service Instance]

Reviewer: You can also use `<<service-instance>>` to cross-link on the same page. I suggest doing so for all glossary terms referring to another term.

////

== General

[[service]]Service::
A service that can be ordered.
It is automatically provisioned, can be consumed by a service consumer and generally adheres to our quality requirements.

[[service-consumer]]Service Consumer::
An application that connects to a service instance as a client.

[[service-order]]Service Order::

> The act of deploying a Kubernetes resource that triggers the provisioning of a new service instance.

Reviewer: Is _Service Order_ a noun or a verb? To me, it feels more like a noun.

```suggestion
A Kubernetes resource defining a service instance and governing the provisioning of a new service instance.
```

[[service-instance]]Service Instance::
A single isolated occurrence of a service.

[[cloud-instance]]Cloud Instance:::
An instance of a cloud service offering by a cloud service provider.

[[converged-service-instance]]Converged Service Instance:::

> An instance of a converged service blueprint.

Reviewer: And what is a _converged service blueprint_?

Either in the same namespace as the service order was made or in another namespace.

== Service Location

Services can be provisioned on various places:

[[converged-cluster]]Converged Cluster::
Service instance runs on the same cluster as the service consumer.

[[service-cluster]]Service Cluster::
Service runs on one or more dedicated clusters, reserved for services.

[[cloud-service]]Cloud Service::
A type of service architecture offered by a cloud service provider such as MS, AWS or GCP.

[[converged-service]]Converged Service::
A type of service developed to run on a converged cluster.

[[converged-service-architecture]]Converged Service Architectures::
Architecture refers to the deployment strategy of a service instance in relation to high availability.

[[standalone]]Standalone:::
A single node instance without any high availability.

[[replicated]]Replicated:::
The service offers data replication to one or more other instances.

[[clustered]]Clustered:::

> A fully clustered version of the service, usually offering active / active replication with fully automated cluster handling.

Author: Is there a reason why you copied the full glossary from AppCat? IMHO the appcat glossary doesn't fully apply to APPUiO Managed Kubernetes Services. So I'd opt to remove that again from this document here.

Reviewer: Yep, that was confusing. Up until now, I had the impression to read an AppCat related documentation.

Reviewer 2: I copied part of AppCat glossary cause I thought we want to align it to APPUiO Managed Kubernetes Services down the road, once we add more information. Sure, I will remove it.

Author: Yes, better get rid of this part again, this confuses too much and doesn't apply for this product.

== Personas

[[End-user]]End User::
A human that interacts with the service (uses it).

[[Technical-VSHNeer]]Technical VSHNeer::
A VSHNeer doing system administration, solution architecture and other technically focused jobs.

> == How-to guides

Reviewer: Looks misplaced. I would not expect a list of how-tos to be listed on a glossary page.

Reviewer 2: This list was already on the glossary page; I just added the header and edited it a bit.. I assume @tobru added it to the glossary page, maybe he can share his opinion as well?

Author: The list was already there, see https://kb.vshn.ch/cloud-kubernetes/references/glossary.html

But the title "How-to guides" was introduced in this (104ed609617c67d4d1789e6007df95a1d060ca9f) commit, which is a mistake.
